### PR TITLE
feat(typeahead): add support for the focusFirst option

### LINKED ISSUE #748: Prevent auto select of first item in typehead
Is it possible to prevent auto select of the first item in the typehead component? I need this because I want the user to be able to press enter if he wants to search normally and only if an item in the list is selected by the user e.g. using the mouse or arrow keys he should navigate to the item.

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Closes #748

### CODE REVIEW

**Comment by wesleycho** [comment]
LGTM

File: src/typeahead/typeahead-config.ts [hunk 1]
```diff
@@ -8,5 +8,6 @@ import {Injectable} from '@angular/core';
 @Injectable()
 export class NgbTypeaheadConfig {
   editable = true;
+  focusFirst = true;
```
Comment (maxokorokov) [inline src/typeahead/typeahead-config.ts]: Looks like you haven't updated `typeahead-config.spec.ts`

**Comment by maxokorokov** [comment]
LGTM, except for the comment

File: src/typeahead/typeahead-config.ts (hunk 1 above)
Comment (pkozlowski-opensource) [inline src/typeahead/typeahead-config.ts]: Thnx @maxokorokov , changed.

**Pull request closed without merging**